map/src/menu/MainMenu.js

```diff
@@ -241,6 +249,13 @@ export default function MainMenu({
         }
     }, [menuInfo]);
 
+    useEffect(() => {
+        const intervalId = setInterval(() => {
+            updateUserRequests(ctx).then();
+        }, 10000);
+        return () => clearInterval(intervalId);
+    }, []);
+
```

Comment:
The 10-seconds interval seems incorrect. What happens if the user clicks to update the request in the middle of a 10-second interval and then closes the page in just 1 second? It seems better to use some debouncer to avoid unnecessary API requests.